### PR TITLE
    TRUNK-5813: Replace all ExpectedException Rules and the @Test(expected..)

### PR DESCRIPTION
**Pull request by achilep**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5813
## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/ConceptServiceTest.java [hunk 1]
```diff
@@ -1251,13 +1246,12 @@ public void saveConceptNameTag_shouldNotSaveAConceptNameTagIfTagExists() {
 		ConceptNameTag secondNameTag = new ConceptNameTag();
 		secondNameTag.setTag(tag);
 		
-		ConceptNameTag existingConceptNameTag = conceptService.saveConceptNameTag(secondNameTag);
-		
+		assertThrows(Exception.class, () -> conceptService.saveConceptNameTag(secondNameTag));
 		assertNull(secondNameTag.getId());
-		assertEquals(existingConceptNameTag.getId(), nameTag.getId());
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/ConceptServiceTest.java:1257]: Why did you remove this line?
From reading it I feel like its supposed to show that the `existingConceptNameTag.getId()` did not change. Thats why I believe it should stay. Unless you convince me it should not :)

Comment (achilep) [reply api/src/test/java/org/openmrs/api/ConceptServiceTest.java:1257]: the exception happen at with this method  :		
       
         ConceptNameTag existingConceptNameTag = conceptService.saveConceptNameTag(secondNameTag);

if it has to stay , it will look like this 

    assertThrows(Exception.class, () ->  assertEquals(conceptService.saveConceptNameTag(secondNameTag).getId(), nameTag.getId());

Comment (teleivo) [reply api/src/test/java/org/openmrs/api/ConceptServiceTest.java:1257]: Thats right but you already wrapped the `conceptService.saveConceptNameTag(secondNameTag)` in an assertThrows. The assertThrows allows to do assertions after it. That is the reason why for example `assertNull(secondNameTag.getId());` is even possible. So it should also be possible to keep `		assertEquals(existingConceptNameTag.getId(), nameTag.getId());` in.

this is how I expect it to look like

```java
	assertThrows(Exception.class, () -> conceptService.saveConceptNameTag(secondNameTag));
	assertNull(secondNameTag.getId());
	assertEquals(existingConceptNameTag.getId(), nameTag.getId());
```

**Comment by achilep** [comment]
@teleivo i try it 
Screenshot from 2020-07-07 20-07-20
The is the output :
Screenshot from 2020-07-07 20-07-53
